tests/e2e_logging.py

```diff
@@ -227,16 +227,22 @@ def test_protocols(network, args):
     expected_response_body = body[: body.rfind("\n")]
 
     # Test additional HTTP versions with curl
-    for (protocol, expected_error) in (
+    for (protocol, expected_errors) in (
         ("", None),
         ("--http1.0", None),
         ("--http1.1", None),
         ("--http2", None),  # Upgrade request is ignored
-        ("--http2-prior-knowledge", "Error in the HTTP2 framing layer"),
-        ("--http3", "the installed libcurl version doesn't support this"),
+        ("--http2-prior-knowledge", ["Error in the HTTP2 framing layer"]),
+        (
+            "--http3",
+            [
+                "the installed libcurl version doesn't support this",
+                "option --http3: is unknown",
+            ],
+        ),
     ):
         res = infra.proc.ccall("curl", protocol, *common_options)
-        if expected_error is None:
+        if expected_errors is None:
```

Comment:
imho, [] is a reasonable way to express "no expected errors", and then this can be "if expected_errors:" but that's clearly nitpicking.

Reply:
or set()

Reply:
Yeah, I think so too, just didn't want the diff to be larger than necessary, will change.